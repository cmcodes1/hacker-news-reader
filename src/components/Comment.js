import React, { useState, useEffect, } from 'react';
import { View, Text, StyleSheet, } from 'react-native';
import { decode } from 'html-entities';
import CommentsService from '../api/CommentsService';
import dateFromNow from '../helpers/DateFromNow';
import colors from '../theme/colors';

export default function Comment({ navigation, by, kids, text, time, }) {

    const commentsService = new CommentsService()

    const [firstReply, setFirstReply] = useState({})

    useEffect(() => {
        kids && fetchFirstReply()
    }, [navigation])

    const fetchFirstReply = async () => {

        console.log('kids[0]', kids[0]);

        const responseData = await commentsService.getFirstReply(kids[0])

        if (responseData.status == 'success') {
            setFirstReply(responseData.payload)
        }

    }

    useEffect(() => {
        console.log('firstReply', firstReply)
    }, [firstReply])

    return (
        <View style={styles.item}>
            <Text style={styles.smallText}>{'\u25B2 '}{by}{' '}{dateFromNow(time)}</Text>
            <Text style={styles.title}>{text}</Text>
            {
                kids && firstReply &&
                <View style={styles.item}>
                    <Text style={styles.smallText}>{'\u25B2 '}{firstReply.by}{' '}{dateFromNow(firstReply.time)}</Text>
                    <Text style={styles.title}>
                        {decode(firstReply.text, { level: 'html5' }).replaceAll("<[^>]*>", "")}
                    </Text>
                </View>
            }
        </View>
    )
}

const styles = StyleSheet.create({
    item: {
        margin: 10,
    },
    row: {
        flexDirection: 'row',
        flexWrap: 'wrap',
    },
    title: {
        fontSize: 14,
        color: colors.textPrimary,
    },
    bigText: {
        fontSize: 14,
    },
    smallText: {
        fontSize: 11,
    },
})