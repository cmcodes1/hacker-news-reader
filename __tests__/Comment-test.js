import React from 'react'
import { create } from 'react-test-renderer'
import Comment from '../src/components/Comment'

jest.useFakeTimers()

const tree = create(<Comment />)

test('snapshot', () => {
    expect(tree).toMatchSnapshot()
})