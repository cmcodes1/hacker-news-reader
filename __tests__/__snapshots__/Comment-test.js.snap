// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "margin": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 11,
      }
    }
  >
    â–² 
     
    on undefined at 12:NaN am
  </Text>
  <View
    style={
      Array [
        Object {},
        undefined,
        undefined,
      ]
    }
    testID="html"
  >
    <View
      style={
        Array [
          Object {},
          undefined,
          undefined,
        ]
      }
      testID="body"
    >
      <Text
        allowFontScaling={true}
        selectable={false}
        style={
          Array [
            Object {
              "fontSize": 14,
            },
            undefined,
            undefined,
          ]
        }
      >
        undefined
      </Text>
    </View>
  </View>
</View>
`;
